js/renovation/ui/scroll_view/scrollable_simulated.tsx

```diff
@@ -344,18 +344,22 @@ export class ScrollableSimulated extends JSXComponent<ScrollableSimulatedPropsTy
     }
 
     this.updateHandler();
+    const isTopDefined = location.top !== undefined;
+    const isLeftDefined = location.left !== undefined;
 
     // TODO: try to simplify it
-    if (this.direction.isVertical) {
+    if (this.direction.isVertical && isTopDefined) {
       const scrollbar = this.vScrollbarRef.current!;
+
       location.top = scrollbar.getLocationWithinRange(
-        location.top! + this.vScrollLocation,
+        location.top as number + this.vScrollLocation,
```

Comment:
The 'location' variable value is returned by the 'restoreLocation' function.
The result of this function is declared as 
```
export interface ScrollOffset {
  top: number;
  left: number;
}
```
Why we need to write another 'as number' cast at these line?

Reply:
restoreLocation returns only part of ScrollOffset:
```typescript
xport function restoreLocation(
  location: number | Partial<{ x: number; y: number; top: number; left: number }>,
  direction?: ScrollableDirection,
): Partial<ScrollOffset> {
```

Reply:
I see.
Then it requires another change.
It will be ready soon.

Reply:
"Partial<ScrollOffset>" is user by a mistake.
Actual result is a "{ left, top }" object.
Both the 'left' ad 'top' can be undefined:
```
    return {
      left: isDefined(left) ? -left : undefined,
      top: isDefined(top) ? -top : undefined,
    };
```

We need this change:
1. at line '342':
if (!location.top && !location.left) -> **if (!isDefined(location.top) || !isDefined(location.left))** 

2. in restore_location.ts:
**export interface Location {
  top: number | undefined;
  left: number | undefined;
}**
export function restoreLocation(
  location: number | Partial<{ x: number; y: number; top: number; left: number }>,
  direction?: ScrollableDirection,
): **Location** 

Some tests can fail, lets extract this change to a new PR.